language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -386,11 +388,20 @@ genDefDataType curPkgId conName mod tpls def =
                             ] ++
                             ["};"]
                         associatedTypes =
-                          maybe [] (\key ->
-                              [ "export namespace " <> conName <> " {"
-                              , "  export type Key = " <> fst (genType (moduleName mod) (tplKeyType key)) <> ""
+                          case tplKey tpl of
+                            Nothing -> []
+                            Just key ->
+                              let tT = conName
+                                  tK = tT <> ".Key"
+                                  tI = "typeof " <> tT <> ".templateId" in
+                              [ "export namespace " <> tT <> " {"
+                              , "  export type Key = " <> fst (genType (moduleName mod) (tplKeyType key))
+                              , "  export type CreateEvent = damlLedger.CreateEvent" <> "<" <> tT <> ", " <> tK <> ", " <> tI <> ">"
+                              , "  export type ArchiveEvent = damlLedger.ArchiveEvent" <> "<" <>  tT <> ", " <> tI <> ">"
+                              , "  export type Event = damlLedger.Event"  <> "<" <>  tT <> ", " <> tK <> ", " <> tI <> ">"
+                              , "  export type Query = damlLedger.Query"  <> "<" <>  tT <> ">"
```

Comment:
There's no big gain from writing `T.Query` instead of `Query<T>`. Thus, let's drop this type synonym.

Reply:
ok